### PR TITLE
Add component for layout

### PR DESCRIPTION
**Pull request by Aritra8438**
# Description

This PR will enable users to **add**, **modify** and **delete layouts**. Users can now
- [x] create layout from the selected case.
- [x] enforce layout to the selected (or all) test cases.
- [x] Copy a layout.
- [x] Delete a layout.
- [x] Modify line information of the layout.

https://github.com/user-attachments/assets/4cefbfb3-6518-49ea-a7b4-7191ebc7514d

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [x] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

File: frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.test.ts [hunk 1]
```diff
@@ -398,7 +398,7 @@ describe('CaseEdit.vue', () => {
     expect(modalInputs.length).toBe(5);
 
     const modalButton = modalBody.find('button[data-array-modal-generate]');
-    expect(modalButton.exists).toBeTruthy();
+    expect(modalButton.exists()).toBeTruthy();
```
Comment (Aritra8438) [inline frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.test.ts:401]: Fixing an error that I made in one of my previous PRs.

**Review by Aritra8438 — COMMENTED** [review]
.

**Comment by Aritra8438** [comment]
Hi @carlosabcs, PTAL!

**Review by carlosabcs — COMMENTED** [review]
Just minor comments

File: frontend/www/js/omegaup/components/problem/creator/cases/LayoutSidebar.vue [hunk 2]
```diff
@@ -0,0 +1,262 @@
+<template>
+  <div class="h-77 overflow-auto">
```
[the remark this answers is not in the document]
Comment (Aritra8438) [reply frontend/www/js/omegaup/components/problem/creator/cases/LayoutSidebar.vue:2]: When there are many `layout`s, the sidebar should be scrollable but the 3 buttons at the bottom should be fixed.

[the remark this answers is not in the document]
Comment (Aritra8438) [reply frontend/www/js/omegaup/components/problem/creator/cases/LayoutSidebar.vue:2]: image

File: frontend/www/js/omegaup/components/problem/creator/cases/LayoutSidebar.vue [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+}
+
+.w-84 {
+  width: 84%;
+}
+
+.table td {
+  vertical-align: middle;
+  border: none;
+}
```
[the remark this answers is not in the document]
Comment (Aritra8438) [reply frontend/www/js/omegaup/components/problem/creator/cases/LayoutSidebar.vue]: Done! @carlosabcs. PTAL!

File: frontend/www/js/omegaup/components/problem/creator/cases/LayoutSidebar.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlosabcs) [reply frontend/www/js/omegaup/components/problem/creator/cases/LayoutSidebar.vue:2]: Makes sense.

File: frontend/www/js/omegaup/problem/creator/modules/cases.ts [hunk 4]
```diff
@@ -268,34 +273,191 @@ export const casesStore: Module<CasesState, RootState> = {
       }
       state = assignMissingPoints(state);
     },
-    addLayoutLine(state) {
-      const payload: CaseLine = {
-        lineID: uuid(),
-        caseID: null,
-        label: 'NEW',
+    addLayoutFromSelectedCase(state) {
+      if (!state.selected.caseID || !state.selected.groupID) {
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/problem/creator/modules/cases.ts:277]: Please add support to each case in the unit tests as CodeCov is complaining about this.

**Review by carlosabcs — COMMENTED** [review]
Complete codecov coverage requirements and this will be approved by my side.

**Comment by Aritra8438** [comment]
@heduenas, PTAL!

**Review by carlosabcs — APPROVED** [review]

**Comment by Aritra8438** [comment]
@heduenas, would you please enable auto-merge again? The PR had merge-conflicts.

**Pull request merged**